tf2_ros/include/tf2_ros/message_filter.h

```diff
@@ -512,10 +519,12 @@ class MessageFilter : public MessageFilterBase, public message_filters::SimpleFi
     rclcpp::Time stamp = mt::TimeStamp<M>::value(*message);
 
     bool transform_available = true;
+    FilterFailureReason error = filter_failure_reasons::Unknown;
     try {
       future.get();
     } catch (...) {
       transform_available = false;
+      error = filter_failure_reasons::OutTheBack;
```

Comment:
I had to double-check, but I think this is always true?

Reply:
what do you mean?
`error=Unknown` will happen in [L538](https://github.com/ros2/geometry2/blob/5f611e972a095b3fa7311724325b434b8f9b9ffe/tf2_ros/include/tf2_ros/message_filter.h#L538) and L547, that's why `OutTheBack` isn't unconditionally used.

IIUC, the ROS 1 code behaved in the same fashion.

Reply:
I meant, if we get an exception is it always because of an "OutTheBack" error? I think this is correct.

Reply:
ah the other exception that can be set is [this one](https://github.com/ros2/geometry2/blob/f4ea23490cbbdde8c18725a458153a5fe5d456e7/tf2_ros/src/buffer.cpp#L244), I really don't understand if it should be considered the same case or not.

Reply:
I think it does, it looks like the only time the callback can fail is this case:

https://github.com/ros2/geometry2/blob/f4ea23490cbbdde8c18725a458153a5fe5d456e7/tf2/src/buffer_core.cpp#L1364-L1366

Which looks like an "OutTheBack" error.